src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -194,7 +197,7 @@ public LowLevelTransaction(LowLevelTransaction previous, TransactionPersistentCo
 
                 _scratchPagerStates = previous._scratchPagerStates;
 
-                _state = previous._state.Clone();
+                _state = previous._state;
```

Comment:
I wonder if we have some notion of "read only" mode of the state (even only in Debug). I'm a bit worried that we reference here the state instance from `StorageEnvironment` so if anyone modifies it directly then read transactions would see it immediately.

Reply:
So basically the state kept in `StorageEnvironment.State` is read-only.

Reply:
That is guaranteed because `StorageEnvironment.State` is only modified under a transaction lock. In fact the transaction lock we want to actually kill with @ayende in the near future.